R/pkg/tests/fulltests/test_mllib_classification.R

```diff
@@ -488,4 +488,36 @@ test_that("spark.naiveBayes", {
   expect_equal(class(collect(predictions)$clicked[1]), "character")
 })
 
+test_that("spark.fmClassifier", {
+  df <- withColumn(
+    suppressWarnings(createDataFrame(iris)),
+    "Species", otherwise(when(column("Species") == "Setosa", "Setosa"), "Not-Setosa")
+  )
+
+  model1 <- spark.fmClassifier(
+    df,  Species ~ .,
+    regParam = 0.01, maxIter = 10, fitLinear = TRUE, factorSize = 3
+  )
+
+  prediction1 <- predict(model1, df)
+  expect_is(prediction1, "SparkDataFrame")
```

Comment:
Can we also check the predict result here?

Reply:
I am not sure if such check are really useful here. In practice fitting is not unlikely failure point and most likely problems are related to parameter passing.

Reply:
I looked other classification tests. It seems other tests checked the ```typeof``` and result of the prediction.  I guess it might be better to be consistent with other tests?

Reply:
`typeof` is not applicable here.  `typeof` is S compatibility thingy, and can be used only to distinguish between core types (here it could only determine if value is S4 type).